### ISSUE TITLE
InviteServerContext calls .setDescription() after session.accept()

### ISSUE THREAD

**Issue by markandrus** [onsip/SIP.js#75]
I have some code that I'm testing in Node. After setting up the UA (SIP.js 0.6.1), I do

``` js
ua.on('invite', function(session) {
  session.reject();
});
```

And then I send an INVITE to my UA. But when my UA receives the INVITE, SIP.js logs

```
Tue Jul 22 2014 14:14:55 GMT-0700 (PDT) | sip.ua | INVITE received but WebRTC is not supported
```

And then I get the following error:

```
Uncaught TypeError: Cannot call method 'receiveResponse' of null
 at Object.IncomingRequest.reply (sip.js/dist/sip.js:4083:27)
 at Object.UA.receiveRequest (sip.js/dist/sip.js:8842:19)
 at Object.Transport.onMessage (sip.js/dist/sip.js:8106:17)
 ...
```

`request.reply(488)` in `UA.prototype.receiveRequest` fails because the `server_transaction` on the `IncomingRequest` is `null`. But really, I'd like to be able to `reject()` the INVITE before SIP.js sends the 488.

**Comment by markandrus** [comment]
Just a followup on this issue: I had resigned myself to this behavior, until I started looking at another feature—the ability to override PeerConnection `iceServers` per-Session.

Because `inviteServerContext.accept` calls `mediaHandler.updateIceServers` with the optional STUN and TURN server arguments after the MediaHandler has been constructed and after `mediaHandler.setDescription` has been called, I needed to defer the actual construction of the PeerConnection and the call to `peerConnection.setRemoteDescription` in the MediaHandler until `inviteServerContext.accept` calls `mediaHandler.getDescription`.

So a custom MediaHandler allowed me to get the behavior I originally described in this ticket, as well as the added benefit of being able to override the PeerConnection `iceServers` per-Session. I would be happy closing this ticket. Thanks!

**Comment by egreenmachine** [comment]
We believe that this issue may be resolved in the latest tagged version [0.8.0](https://github.com/onsip/SIP.js/releases/tag/0.8.0). Can you please retest with that version and let us know if this is still an issue. We will close this issue on 11/1/17 if we do not hear back. Thanks.

**Comment by egreenmachine** [comment]
As stated in previous post, closing due to inactivity and proposed solution in 0.8.X. If this turns out to still be an issue, please re-open.
